### ISSUE TITLE
Error when executing `shards` a second time with a Projectfile dependency

### ISSUE THREAD

**Issue by luislavena** [ysbaddaden/shards#60]
Hello,

This issue is related to #52, and seems to occur not when installing dependencies, but subsequent commands runs against the generated lockfile.

The following `shard.yml` is able to reproduce it:

```yaml
name: testing
version: 0.0.1

dependencies:
  routing:
    github: bcardiff/crystal-routing
    branch: master
```

On first run (no `.shards`, `libs` or `shard.lock` exist):

```
$ shards --version
Shards 0.5.0 [a501f92] (2015-09-28)
$ shards install
Updating https://github.com/bcardiff/crystal-routing.git
Installing routing (master)
```

On second run:

```
$ shards install
Updating https://github.com/bcardiff/crystal-routing.git
in shard.yml:1: Shards::ParseError: missing required attribute: version at 1:0

  1. name: routing
     ^
```

On removal of `libs`:

```
$ shards install
Updating https://github.com/bcardiff/crystal-routing.git
Installing routing (2289be7382ee027c0905e5c56523356345e46541)
```

And second run:

```
$ shards install
Updating https://github.com/bcardiff/crystal-routing.git
in shard.yml:1: Shards::ParseError: missing required attribute: version at 1:0

  1. name: routing
     ^
```

Notice the difference between the first install (pointed to `master` and the second (when using the commit from `shard.lock`)

However, the error around missing `version` cannot be avoided unless `.shards` directory is removed and dependency cloned again, at which point the issue repeats again.

This also happens when invoking `check` too:

```
$ rm -rf .shards/ libs/

$ shards check
Dependencies aren't satisfied. Install them with 'shards install'

$ shards install
Updating https://github.com/bcardiff/crystal-routing.git
Installing routing (2289be7382ee027c0905e5c56523356345e46541)

$ shards check
in shard.yml:1: Shards::ParseError: missing required attribute: version at 1:0

  1. name: routing
     ^
```

This only happens with legacy dependency (Projectfile).

Thank you.

**Comment by ysbaddaden** [comment]
Thanks for the detailed bug report! I'll have a look to understand why the version ends up missing again.

**Comment by bararchy** [comment]
I'm having a similar issue, though my error is a little bit different: 

```
in shard.yml:7: Shards::ParseError: expected MAPPING_START but was SCALAR at 7:12

  1. 
  2. dependencies:
  3.     github: manastech/crystal-toml
                ^


```

**Comment by jhass** [comment]
@bararchy I don't think this is similar, it should be:

```yml
dependencies:
  toml:
    github: manastech/crystal-toml
    branch: master

**Comment by bararchy** [comment]
@jhass I changed it, thanks, now I get the error XD 

```
Updating https://github.com/manastech/crystal-toml.git
in shard.yml:1: Shards::ParseError: missing required attribute: version at 1:0

  1. name: toml
     ^


```

**Comment by jhass** [comment]
So add a version.

**Comment by ysbaddaden** [comment]
@bararchy I'm working on that (this is the issue here). The difficulty is dealing with projects that don't have a `shard.yml` yet, or don't even have version tags.

**Comment by bararchy** [comment]
@ysbaddaden will adding a random version fix that ?

**Comment by ysbaddaden** [comment]
@bararchy @luislavena this should be fixed by e4dbbdb please reopen the issue if it doesn't. You shouldn't have to remove the `libs` folder, but if it fails, please try to remove it first.

Thanks again for debugging all these issues! I'll make a v0.5.1 patch release quickly once these bugs are smashed.

**Comment by luislavena** [comment]
@ysbaddaden works perfectly, thank you :heart: :heart: :heart:
